### PR TITLE
feat: Use activity detection handler for motion changes

### PR DESCRIPTION
**Pull request by paultranvan**
We used to rely on the activity stored in locations points to send
motion_activity to the server. However, we noticed the activity is not
always specified in the points, which was causing bad section
segmentation.

Now, we rely on the activity change API: each time an activity is
detected, we store it locally. Then, at upload time, we extract the
relevant activities (i.e. timestamp lower than the last location point,
and avoid activity duplicates), and remove it afterwards.

### CODE REVIEW

File: src/app/domain/geolocation/tracking/storage.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    const activitiesBeforeTs = activities.filter(activity => {
+      return activity?.data?.ts <= beforeTs
+    })
+    return activitiesBeforeTs
+  }
+  return activities
+}
+
+export const removeActivities = async ({ beforeTs } = {}) => {
+  console.log('remove activities')
```
Comment (zatteo) [inline src/app/domain/geolocation/tracking/storage.js]: If you want to keep it, use the `Log` which uses cozy-minilog logger

File: src/app/domain/geolocation/tracking/index.js [hunk 2]
```diff
@@ -120,6 +121,11 @@ export const setTrackingConfig = async newTrackingConfig => {
   await storeData(StorageKeys.GeolocationTrackingConfig, newTrackingConfig)
 }
 
+export const handleActivityChange = async event => {
+  Log('[ACTIVITY CHANGE] - ' + JSON.stringify(event))
```
Comment (zatteo) [inline src/app/domain/geolocation/tracking/index.js:125]: These log is duplicated with the one from the headless task no ?

`Log('[ACTIVITYCHANGE] -' + JSON.stringify(params))`

File: src/app/domain/geolocation/tracking/storage.js (hunk 1 above)
Comment (paultranvan) [reply src/app/domain/geolocation/tracking/storage.js]: Yep sorry, the rebase was not perfect

File: src/app/domain/geolocation/tracking/index.js (hunk 2 above)
Comment (paultranvan) [reply src/app/domain/geolocation/tracking/index.js:125]: Good catch :+1:

File: src/app/domain/geolocation/tracking/tracking.js [hunk 3]
```diff
@@ -247,14 +271,21 @@ const addPoint = (content, point, filtered) => {
...
-    content.push(motionActivity)
+const getActivityMode = activity => {
+  const modeKeys = [
+    'cycling',
+    'running',
+    'walking',
+    'automotive',
+    'stationary',
+    'unknown'
+  ]
```
Comment (zatteo) [inline src/app/domain/geolocation/tracking/tracking.js:282]: You can move this constant outside the function.

**Comment by zatteo** [comment]
Nice refacto with the storage 👍

**Review by zatteo — APPROVED** [review]

File: src/app/domain/geolocation/tracking/tracking.js (hunk 3 above)
Comment (paultranvan) [reply src/app/domain/geolocation/tracking/tracking.js]: Fair enough, I move it

**Pull request merged**